e2e/testinfra/terraform/common/service_accounts.tf

```diff
@@ -102,3 +102,15 @@ resource "google_project_iam_member" "ns-reconciler-wi-sa-iam" {
   member  = "serviceAccount:${data.google_project.project.project_id}.svc.id.goog[config-management-system/ns-reconciler-test-ns]"
   project = data.google_project.project.id
 }
+
+resource "google_service_account" "e2e_log_writer_sa" {
```

Comment:
hmm. Do we need this? 

I thought the post-sync deployment just writes to stdout and the node's fluentd daemon writes it to Cloud Logging.

Reply:
IMO we should instead have Cloud Logging reader granted to the e2e test account, so it can query Cloud Logging to validate the e2e behavior works. But I guess you've decided to just read the pod logs with kubectl instead.

Reply:
FWIW, if the prow node is sending a lot of logs, the logs you want to test could disappear from the node log cache before you can read them. This seems unlikely, but hypothetically possible. It's also possible the logging happens on one node and then the pod gets evicted rescheduled to another node, especially on autopilot, which could also cause log reading from kubectl to fail.

Querying Cloud Logging would prevent both of those edge cases, since it caches for 30 days at least.

Reply:
Ooo. I see you ARE already querying Cloud Logging in the e2e tests. 

How does the prow job get log reader?

Reply:
Forgot to remove this, the test no longer requires a writer permission setup. Log reader is created in the postsync_logging_test using default application creds, no GSA is required. The Cloud Logging related tests has option requireGKE enabled, to separate from the presubmit(prow) tests. The later ones only check for podlogs, not accessing Logging API.